### DISCUSSION TITLE
AWS Lambda support

### DISCUSSION THREAD

**Discussion by schonert** [honojs/hono#916] [Q&A]
Hey there!

I've been following this project for quite some time now - crossing my fingers that I'd get the opportunity to play with it. I find myself returning with the same question - how does it play with AWS lambdas, and could I get rid of some of the tedious headaches? 

Altho I always leave empty-handed I keep having this growing feeling that I might have simply overseen it, so I eventually come back.

Could someone enlighten me and relieve me of my endless search?
- Should I consider using Hono to wrap and manage my Lambda handlers?
- If so - should we go with a main app handler or just create an Hono app for each?

Appreciate the help!

**Comment by yusukebe** [comment]
Hi @schonert !

I am not familiar with AWS Lambda, but [Hono runs on Node.js with the Node.js Adapter](https://hono.dev/getting-started/nodejs), so that might work. And this may be too early for practical use, but there is an example of running Bun on AWS Lambda and Hono run on it.

https://twitter.com/jarredsumner/status/1624596087662333954

If there are any examples of Hono running on AWS Lambda, I would like to know too.

**Comment by yusukebe** [comment]
Wow, "bun-lambda" is out.

https://twitter.com/jarredsumner/status/1628606016517726208
https://github.com/oven-sh/bun/tree/main/packages/bun-lambda#readme

**Comment by yusukebe** [comment]
I have tried to run Hono with AWS Lambda. It is possible, but it was hard things mapping Lamda's handler with Hono's, so I decided not to do it. I wonder if anyone would implement it.

**Reply by schonert** [reply]
There are projects that do the mapping for the express etc. we could rely on them for the heavy lifting. One example here https://github.com/dougmoscrop/serverless-http

**Comment by yusukebe** [comment]
Short code, but it works!

https://gist.github.com/yusukebe/ed1fc4f058487729fcfa2581adc2abf4

**Comment by yusukebe** [comment]
Hi! I've created the PR for supporting AWS Lambda!

https://github.com/honojs/hono/pull/987
